omnigan/data.py

```diff
@@ -66,6 +118,90 @@ def decode_segmap(image, nc=19):
     return rgb
 
 
+def find_closest_class(pixel, dict_classes):
+    """Takes a pixel as input and finds the closest known pixel value corresponding to a class in dict_classes
+    Arguments:
+        pixel -- tuple pixel (R,G,B,A)
+    Returns:
+        tuple pixel (R,G,B,A) corresponding to a key (a class) in dict_classes
+    """
+    min_dist = float("inf")
+    closest_pixel = None
+    for pixel_value in dict_classes.keys():
+        dist = np.sum(np.absolute(np.subtract(pixel, pixel_value)))
```

Comment:
yes, right, I don't know what I was thinking... but maybe Euclidean distance is better ? 
That being said, I doubt there will actually be that significant changes so that you really have a random color that doesn't correspond to anything, right ?
 I guess an extra thing would then be to look at neighbouring classes to decide in the case of a tie but that seems like an overkill, so I don't think you need to do that!

Reply:
Yes I guess I would just put Euclidian distance for the sake of geometers :)